youtube_dl/extractor/tistory.py

```diff
@@ -0,0 +1,228 @@
+    compat_urllib_error,
+    compat_urlparse,
+    compat_str
+)
+
+import os.path
+import cgi
+import re
+
+
+class TistoryBaseIE(InfoExtractor):
+    _TI_MEDIA_URL = r'https?://cfile[0-9]*.uf.(tistory.com|daum.net)/(?:media|attach|attachment|original)/(?P<id>[A-Za-z0-9]*)'
+
+    def _ti_unquote(self, url):
+        return compat_urlparse.unquote(url)
+
+    def _ti_get_title(self, url, response):
+        _, params = cgi.parse_header(response.info().get('Content-Disposition', ''))
+        if "filename" in params:
+            filename = params["filename"]
+        else:
+            filename = url_basename(url)
+
+        retval = os.path.splitext(self._ti_unquote(filename))[0]
+
+        if type(retval) != compat_str:
+            retval = retval.decode('UTF-8')
+
+        return retval
+
+    def _ti_get_ext(self, mime):
+        ext = mimetype2ext(mime)
+        if ext == "x-shockwave-flash":
+            ext = "flv"
+        return ext
+
+    def _ti_get_real_from_check(self, check):
+        checkmatch = re.search("(?P<host>(tistory.com|daum.net)).*(?P<server>cfile[0-9]*.uf)@(?P<id>[A-Z0-9]*)(?:\.(?P<ext>[A-Za-z0-9]*))?", check)
+        if not checkmatch:
+            return None
+
+        host = checkmatch.group("host")
+        cfile = checkmatch.group("server")
+        url = checkmatch.group("id")
+        ext = None
+
+        if len(checkmatch.groups()) > 3:
+            ext = checkmatch.group("ext")
+
+        return ("http://" + cfile + "." + host + "/attach/" + url, ext)
+
+    def _ti_get_video_id(self, url):
+        if '_TI_MEDIA_URL_RE' not in self.__dict__:
+            self._TI_MEDIA_URL_RE = re.compile(self._TI_MEDIA_URL)
+        m = self._TI_MEDIA_URL_RE.match(url)
+        assert m
+        return m.group('id')
+
+    def _ti_get_headers(self, url, video_id):
+        self.to_screen('%s: Downloading headers' % (video_id))
+        req = HEADRequest(url)
+
+        return compat_urllib_request.urlopen(req)
+
+    def _ti_detect_swf(self, head):
+        content_type = head.info().get("content-type")
+        content_length = int(head.info().get("content-length"))
+
+        if content_type == "application/x-shockwave-flash" and content_length < 200000:
+            return True
+
+        return False
+
+    def _ti_detect_xml(self, head):
+        content_type = head.info().get("content-type")
+
+        if "xml" in content_type or content_type == "text/html":
+            return True
+
+        return False
+
+    def _ti_get_media(self, url, video_id, head, ext=None, title=None):
+        if head:
+            content_type = head.info().get("content-type")
+            ext = self._ti_get_ext(content_type)
+            title = self._ti_get_title(url, head)
+
+        if not title:
+            title = video_id
+
+        return {
+            "id": compat_str(video_id),
+            "url": url,
+            "title": title,
+            "ext": ext
+        }
+
+    def _ti_read_swf(self, url, video_id, head):
```

Comment:
Lots of codes in this method duplicates swfinterp.py. Re-use existing codes instead.